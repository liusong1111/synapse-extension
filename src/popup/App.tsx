import * as React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Address from '../pages/Main'
import Popup from './Popup'
import AppBar from '../Components/AppBar/index'

export default function App() {
  return (
    <Router>
      <div>
        <AppBar />
        <nav>
          <ul>
            <li>
              <Link to="/">Popup</Link>
            </li>
            <li>
              <Link to="/address">Address</Link>
            </li>
          </ul>
        </nav>
        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Switch>
          <Route path="/address">
            <Address />
          </Route>
          <Route path="/">
            <Popup />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}
