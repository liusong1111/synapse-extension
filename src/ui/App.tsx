import * as React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Address from './pages/Address'
import ImportMnemonic from './pages/ImportMnemonic'
import AppBar from './Components/AppBar/'

export const AppContext = React.createContext({network: 'testnet'});

export default function App() {
  const [network, setNetwork] = React.useState('testnet');

  const handleNetworkChange = (value: string) => {
    setNetwork(value);
  };

  return (
    <Router>
      <AppContext.Provider value={{
        network
      }}>
        <div>
          <AppBar handleNetworkChange={handleNetworkChange} />
          <nav>
            <ul>
              <li>
                <Link to="/address">Address</Link>
              </li>
              <li>
                <Link to="/import-mnemonic">Import Mnemonic</Link>
              </li>
            </ul>
          </nav>
          {/* A <Switch> looks through its children <Route>s and
              renders the first one that matches the current URL. */}
          <Switch>
            <Route path="/address">
              <Address />
            </Route>
            <Route path="/import-mnemonic">
              <ImportMnemonic />
            </Route>
          </Switch>
        </div>
      </AppContext.Provider>
    </Router>
  );
}
