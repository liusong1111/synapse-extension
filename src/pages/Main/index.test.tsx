import * as React from 'react';
import { MemoryRouter } from 'react-router-dom'
import App from './index';
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import * as chrome from "sinon-chrome";

describe('React testing library', () => {
  let tree
  beforeEach(() => {
    tree = render(
      <MemoryRouter initialEntries = {['/address']}>
        <App />
     </MemoryRouter>
    );
  });
  afterEach(cleanup)

  beforeAll(() => {
    window.chrome = chrome
  })

  it('should render title', async() => {
    const {
      getByTestId,
      container
    } = tree

    const title = getByTestId('address-title')
    expect(container).toContainElement(title)
    expect(title).toHaveTextContent('Address')
  })

  it('should render address', async() => {
    const {
      getByTestId,
      container
    } = tree

    const address = getByTestId('address-info')
    expect(container).toContainElement(address)
    expect(address).toHaveTextContent(/ck|loading|null|undefined|/)
  })

  it("should render balance", ()=>{
    const {
      getByTestId,
      container
    } = tree

    const balance = getByTestId('balance')
    expect(container).toContainElement(balance)
  })

  it("should render receive / send btn", ()=>{
    const{
      getByTestId,
      container,
    }= tree

    const receiveBtn = getByTestId('receive')
    const sendBtn = getByTestId('send')
    expect(container).toContainElement(receiveBtn)
    expect(container).toContainElement(sendBtn)

  });

});

