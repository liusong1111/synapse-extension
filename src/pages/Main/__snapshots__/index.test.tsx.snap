// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render correctly 1`] = `
<div
  className="makeStyles-container-1"
>
  <default_1
    testId="address-title"
    title="Address"
  />
  <div
    className="address"
  >
    address info
  </div>
</div>
`;
